core/App/utils/helpers.ts

```diff
@@ -28,6 +30,51 @@ export function hashToRGBA(i: number) {
   return '#' + '00000'.substring(0, 6 - colour.length) + colour
 }
 
+export function formatTime(time: Date, translate: TFunction, params?: { long?: boolean; format?: string }): string {
+  const getMonthKey = 'MMMM'
+  const momentTime = moment(time)
+  const monthKey = momentTime.format(getMonthKey)
+  const customMonthFormatRe = /M+/
+  const long = params?.long
+  const format = params?.format
+
+  let formatString = translate('Date.ShortFormat')
+  if (format) {
+    formatString = format
+  } else {
+    if (long) {
+      formatString = translate('Date.LongFormat')
+    }
+
+    // if translation fails
+    if (formatString === 'Date.ShortFormat' || formatString === 'Date.LongFormat') {
+      formatString = 'MMM D, YYYY'
+    }
+  }
+  const customMonthFormat = formatString.match(customMonthFormatRe)?.[0]
+
+  let formattedTime = momentTime.format(formatString)
+
+  if (customMonthFormat) {
+    let monthReplacement = ''
+    const monthReplacementKey = momentTime.format(customMonthFormat)
+    if (customMonthFormat.length === 3) {
```

Comment:
[Magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))? Consider naming it with the rest of the consts at the top.